### ISSUE TITLE
Paper implementation

### ISSUE THREAD

**Issue by ehfo0888** [Janspiry/Palette-Image-to-Image-Diffusion-Models#47]
Hello,

Thank you very much for sharing your code.

when I studied the paper I came across 2 implementation detail. I see that the setting you choose are the same as the hyper parameter for the inpainting?
![Screenshot from 2022-10-03 20-11-20](https://user-images.githubusercontent.com/114948566/193649999-86e7f6c4-d288-426d-b83a-7c67ae20e969.png)

is there an option in the code for regular training the one with 1024 batch size? what is the difference between option 1 and 2 in the screen above?

**Comment by Janspiry** [comment]
Hi, although the implementation uses the same learning rate, the batch_size is far from 1000, so maybe it would be better to implement it conditionally (you can adjust the batch_size option in the configuration).
The second question I did not understand too well ?

**Comment by ehfo0888** [comment]
In the training details it mentioned the number of training step is 1M but in the Diffusion Hyper Parameter section it says only 2000 training. I am confused, what are the differences?

**Comment by Janspiry** [comment]
2000 is time-step of diffusion \alpha, not the  training step

**Comment by ehfo0888** [comment]
Thank you for your time.

1 - how can I increase the number of training steps in the code? 
2- what is the mask you used in the pre-trained model? is it hybrid? I run the colab and changed the type of mask during the testing to free-form but the model couldn't fill the edges? how do you think I should change the setting to cover this issue?

**Comment by Janspiry** [comment]
1. Change the n_iter: https://github.com/Janspiry/Palette-Image-to-Image-Diffusion-Models/blob/136b29f58d0af6e5db9f3655d2891f5a855fcdaa/config/inpainting_celebahq.json#L137
2. hybrid, the unmasked region have changed, Is there any change somewhere else?
![image](https://user-images.githubusercontent.com/32922494/193785402-c970a707-a970-488e-ad8b-061e240d83b7.png)

**Comment by ehfo0888** [comment]
The number of iterations when running the training script is way less than that with the current setting which is 9332.

The only change I did is to change the mask during the test in the path file from center to free form and changed the "n_step" In the test from 1000 to 3000. 
It seems even though I change the mask type it still uses a centering mask!!

What is the correct way of setting a mask for such a picture? and what type of mask do you recommend?

**Comment by Janspiry** [comment]
Did you change these in notebook?

**Comment by hodfa840** [comment]
Yes, I did.
for some reason the notebook doesn't work for irregular mask.
and I have a couple of questions: 
has the "free-form" mask implemented? does free-form mean the user can give any type of corrupted image as an input?
second, Is there a way to save all the images (ie. 1000) diffusion steps I would like to mage a gif from them.

**Comment by Janspiry** [comment]
Yes, you can find it in https://github.com/Janspiry/Palette-Image-to-Image-Diffusion-Models/blob/136b29f58d0af6e5db9f3655d2891f5a855fcdaa/data/util/mask.py#L232, and you need to give the mask manually.
Get more images by
https://github.com/Janspiry/Palette-Image-to-Image-Diffusion-Models/blob/136b29f58d0af6e5db9f3655d2891f5a855fcdaa/config/inpainting_celebahq.json#L71
